src/Atlas2DSlice.ts

```diff
@@ -1,23 +1,148 @@
-import { rayMarchingShaderUniforms } from "./constants/volumeRayMarchShader";
+import { Box3, Box3Helper, BufferGeometry, Color, Group, LineBasicMaterial, Material, Matrix4, Mesh, PlaneGeometry, ShaderMaterial, ShapeGeometry, Vector2, Vector3 } from "three";
+import { Channel, Volume } from ".";
+import { sliceFragmentShaderSrc, sliceShaderUniforms, sliceVertexShaderSrc } from "./constants/volumeSliceShader";
+import { VolumeRenderImpl } from "./VolumeRenderImpl";
+import { SettingsFlags, VolumeRenderSettingUtils, VolumeRenderSettings, defaultVolumeRenderSettings } from "./VolumeRenderSettings";
+import FusedChannelData from "./FusedChannelData";
+import { FuseChannel } from "./types";
+import { ThreeJsPanel } from "./ThreeJsPanel";
+
+const BOUNDING_BOX_DEFAULT_COLOR = new Color(0xffff00);
 
 /**
  * Creates a plane that renders a 2D XY slice of volume atlas data.
  */
-export default class Atlas2DSlice extends RayMarchedAtlasVolume {
-  constructor(volume: Volume) {
-    super(volume);
+export default class Atlas2DSlice implements VolumeRenderImpl {
+  private settings: VolumeRenderSettings | undefined;
+  public volume: Volume;
+  private geometry: ShapeGeometry;
+  protected geometryMesh: Mesh<BufferGeometry, Material>;
+  private geometryTransformNode: Group;
+  private boxHelper: Box3Helper;
+  private uniforms: ReturnType<typeof sliceShaderUniforms>;
+  private channelData: FusedChannelData;
+
+  constructor(volume: Volume, settings?: VolumeRenderSettings) {
+    this.volume = volume;
+    this.uniforms = sliceShaderUniforms();
+    [this.geometry, this.geometryMesh] = this.createGeometry(this.uniforms);
+
+    this.boxHelper = new Box3Helper(
+      new Box3(new Vector3(-0.5, -0.5, -0.5), new Vector3(0.5, 0.5, 0.5)),
+      BOUNDING_BOX_DEFAULT_COLOR
+    );
+    this.boxHelper.updateMatrixWorld();
+    this.boxHelper.visible = false;
+
+    this.geometryTransformNode = new Group();
+    this.geometryTransformNode.name = "VolumeContainerNode";
+
+    this.geometryTransformNode.add(this.boxHelper, this.geometryMesh);
+
+    this.setUniform("ATLAS_X", volume.imageInfo.cols);
+    this.setUniform("ATLAS_Y", volume.imageInfo.rows);
+    this.setUniform("textureRes", new Vector2(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height));
+    this.setUniform("SLICES", volume.z);
     this.setUniform("Z_SLICE", Math.floor(volume.z / 2));
+
+    this.channelData = new FusedChannelData(volume.imageInfo.atlas_width, volume.imageInfo.atlas_height);
+
+    if (!settings) {
+      settings = defaultVolumeRenderSettings();
+      VolumeRenderSettingUtils.resizeWithVolume(settings, volume);
+    }
+    this.updateSettings(settings, SettingsFlags.ALL);
+  }
+
+  public updateSettings(newSettings: VolumeRenderSettings, _dirtyFlags?: number | SettingsFlags) {
+    if (_dirtyFlags === undefined) {
+      _dirtyFlags = SettingsFlags.ALL;
+    } else if (_dirtyFlags === SettingsFlags.NONE) {
+      return;
```

Comment:
when would this happen?

Reply:
The only use case I could see if maybe setting values without restarting the rendering process in PathTracedVolume... but anytime the settings are being changed it should be changing the rendering process and restarting anyways.

Removed this and `SettingsFlags.NONE`!